pkg/controllers/repositoryserver/suite_test.go

```diff
@@ -15,58 +15,174 @@
 
-	k8sClient, err := client.New(cfg, client.Options{Scheme: scheme.Scheme})
+	go func() {
+		err = mgr.Start(ctrl.SetupSignalHandler())
+		c.Assert(err, IsNil)
+	}()
+
+	ns := &v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "repositoryservercontrollertest-",
+		},
+	}
+	ctx := context.Background()
+	cns, err := s.kubeCli.CoreV1().Namespaces().Create(ctx, ns, metav1.CreateOptions{})
 	c.Assert(err, IsNil)
-	c.Assert(k8sClient, NotNil)
+	s.repoServerControllerNamespace = cns.Name
+	s.createRepositoryServerSecrets(c)
 }
 
-func (s *ControllerSuite) TearDownSuite(c *C) {
-	if s.testEnv != nil {
-		c.Log("Tearing down the test environment")
-		err := s.testEnv.Stop()
-		c.Assert(err, IsNil)
+func (s *RepoServerControllerSuite) createRepositoryServerSecrets(c *C) {
+	repoServerUserAccessSecretData := map[string][]byte{
+		"localhost": []byte(DefaultKopiaRepositoryServerAccessPassword),
 	}
+	repoServerAdminSecretData := map[string][]byte{
+		"username": []byte(DefaulKopiaRepositoryServerAdminUser),
+		"password": []byte(DefaultKopiaRepositoryServerAdminPassword),
+	}
+	repoPasswordSecretData := map[string][]byte{
+		repoPasswordKey: []byte(DefaultKopiaRepositoryPassword),
+	}
+	kopiaTLSSecretData, err := getKopiaTLSSecret()
+	c.Assert(err, IsNil)
+	s.repoServerSecrets = repositoryServerSecrets{}
+	s.repoServerSecrets.serverUserAccess, err = s.createSecret("test-repository-server-user-access-", "", repoServerUserAccessSecretData)
+	c.Assert(err, IsNil)
+	s.repoServerSecrets.serverAdmin, err = s.createSecret("test-repository-server-admin-", "", repoServerAdminSecretData)
+	c.Assert(err, IsNil)
+	s.repoServerSecrets.repositoryPassword, err = s.createSecret("test-repository-password-", "", repoPasswordSecretData)
+	c.Assert(err, IsNil)
+	s.repoServerSecrets.serverTLS, err = s.createSecret("test-tls-", v1.SecretTypeTLS, kopiaTLSSecretData)
+	c.Assert(err, IsNil)
+	s.repoServerSecrets.storage, err = s.createSecret("test-repository-server-storage-", "", getDefaultS3StorageLocation())
+	c.Assert(err, IsNil)
+	s.repoServerSecrets.storageCredentials, err = s.createSecret("test-repository-server-storage-Creds-", "secrets.kanister.io/aws", getDefaultS3StorageCreds())
+	c.Assert(err, IsNil)
 }
 
-func (s *ControllerSuite) TestWatch(c *C) {
-	// We give it a few seconds complete it's scan. This isn't required for the
-	// test, but is a more realistic startup scenario.
-	time.Sleep(5 * time.Second)
+func (s *RepoServerControllerSuite) createSecret(name string, secrettype v1.SecretType, data map[string][]byte) (se *v1.Secret, err error) {
+	secret := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: name,
+		},
+		Data: data,
+	}
+	if secrettype != "" {
+		secret.Type = secrettype
+	}
+
+	se, err = s.kubeCli.CoreV1().Secrets(s.repoServerControllerNamespace).Create(context.Background(), secret, metav1.CreateOptions{})
+	return
+}
+
+func (s *RepoServerControllerSuite) TestRepositoryServerStatusIsServerReady(c *C) {
```

Comment:
do we just want to test if the the RespoServer CR's status gets to ready state after sometime of creation or do we want to test anything else as part of this PR/suite?

Reply:
Yes adding more